src/maintainer/conda_forge_yml.rst

```diff
@@ -79,6 +80,34 @@ mapping for Azure-specific configuration options. For example:
       # built packages) as an Azure pipeline artifact that can be downloaded
       store_build_artifacts: False
 
+.. _self-hosted_azure-config:
+
+self-hosted azure
+-----------------
+Conda-Smithy is capable of doing things beyond CF's purposes, it also allows setting up
+self-hosted agents for non-CF infrastructures. Here is an example demonstrating that:
+
+.. code-block:: yaml
+
+      azure:
+        user_or_org: nsls2forge
+        project_name: nsls2forge
+        for local GPU runners
+        settings_linux:
+        pool:
+        name: BNL-machines
+        demands:
+          - gpu -equals true
+        workspace:
+          clean: all
+        strategy:
+          maxParallel: 2
+          variables:
+            cutensor_path: /home/ci_agent/libcutensor_1.0.1/
```

Comment:
Need to fix indentation and clean up some useless keys. This is a better example from the original PR https://github.com/conda-forge/conda-smithy/pull/1296#issue-409225201:
```suggestion
      azure:
        settings_linux:
          pool:
            name: your_local_pool_name
            demands:
              - some_key -equals some_value
          workspace:
            clean: all
          strategy:
            maxParallel: 1
```